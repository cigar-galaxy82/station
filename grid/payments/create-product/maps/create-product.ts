/* eslint-disable jest/no-export */

import { describe, expect } from '@jest/globals';
import { SuperfaceTest } from '@superfaceai/testing';

export async function createProduct(provider: string): Promise<string> {
  const superfacePublishingProfiles = new SuperfaceTest({
    profile: 'payments/create-product',
    provider,
  });
  const result = await superfacePublishingProfiles.run({
    useCase: 'CreateProduct',
    input: {
      name: 'Temporary Product',
      description: 'Used just for testing.',
      type: 'physical',
    },
    testName: 'payments/create-product/temporary',
  });

  return (result.unwrap() as { id: string }).id;
}

export function createProductTest(providerName: string): void {
  describe(`payments/create-product/${providerName}`, () => {
    let superface: SuperfaceTest;

    beforeEach(() => {
      superface = new SuperfaceTest({
        profile: 'payments/create-product',
        provider: providerName,
        testInstance: expect,
      });
    });

    describe('CreateProduct', () => {
      describe('when all inputs are correct', () => {
        it('creates a product', async () => {
          const result = await superface.run({
            useCase: 'CreateProduct',
            input: {
              name: 'Superface milk',
              description:
                'Milk that gives you super-powers, delivered to your door.',
              type: 'physical',
            },
          });
          expect(() => result.unwrap()).not.toThrow();
          expect(result.unwrap()).toMatchSnapshot();
        });
      });
    });
  });
}
