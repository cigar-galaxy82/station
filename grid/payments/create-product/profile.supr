name = "payments/create-product"
version = "1.0.0"

"""
Create product  usecase
"""

usecase CreateProduct unsafe {
  input {
    """
    Name
    Name of the product to be created.
    """
    name! string!

    """
    Description
    Description of the product to be created.
    """
    description! string!

    """
    Product type
    Indicates whether the product is physical or intangible (e.g. a service, or digital goods).

    Possible values:
      - `physical`
      - `digital`
      - `service`
    """
    type enum {
      physical,
      digital,
      service
    }
  }

  result {
    """
    ID
    Unique identifier of the product.
    """
    id! string!
    """
    Product URL
    A URL of the product's publicly-accessible webpage.
    """
    url string
  }

  error {
    """
    Title
    A short, human-readable summary of the problem type.
    """
    title! string

    """
    Detail
    A human-readable explanation specific to this occurrence of the problem.
    """
    detail string
  }

  example Successful {
    input {
      name = "Superface milk"
      description = "Milk that gives you super-powers, delivered to your door."
    }

    result {
      id = "PROD-1192810198"
      url = "https://www.example.com/shop/PROD-1192810198"
    }
  }
}