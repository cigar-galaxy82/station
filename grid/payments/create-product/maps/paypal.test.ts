import { createProductTest } from './create-product';

createProductTest('paypal');
