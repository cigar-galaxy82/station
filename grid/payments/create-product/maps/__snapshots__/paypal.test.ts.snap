// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`payments/create-product/paypal CreateProduct when all inputs are correct creates a product 1`] = `
Object {
  "id": "PROD-4N174200XH811150C",
  "url": undefined,
}
`;
